### PR TITLE
Re-adds comments to the security records console, plus some other records changes

### LINKED ISSUE #11316: Security record consoles no longer have comments
Comments are gone, this is a travesty!

Important notes is fucking useful, comments were important.

### LINKED ISSUE #11723: Sec filing cabinet and pai sec records don't properly list crimes
### Severity

Inconvenience

### Short Description

The sec filing cabinet and pai sec records don't properly turn the crimes list into html

### Reproduction

1. Spawn in a `/obj/structure/filingcabinet/security`
2. Open it
3. See it looks like this:

image

### Round ID

All for a while

### Test Merges

N/A

### BYOND Username

adamsogm

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by adamsong**
# General Documentation

### Intent of your Pull Request

Re-adds comments to the security records console that were not implemented with the last UI update, while also redoing the comment storage internally to work better.
Combined major and minor crime categories due to the lack of need for the distinction, and the need for UI space for the comments. 
Fixed the security filing cabinet and pAI security records not working properly.

<details>
<summary>Image</summary>

image

</details>

### Why is this change good for the game?
A lot of people like comments, and wanted them added back, this does that.

# Changelog

Fixes #11723 
Fixes #11316 
Adds feature https://forums.yogstation.net/threads/if-security-records-do-not-get-comments-back-then-please-revert-the-ui-change.23664/

:cl:  
rscadd: Added security comments to security records console
bugfix: fixed security filing cabinet
bugfix: fixed pAI security records
tweak: tweaked how security comments are stored 
tweak: removed the distinction between major and minor crimes in the sec records
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/security.dm [hunk 1]
```diff
@@ -627,39 +617,39 @@
...
 					if(istype(active_general_record, /datum/data/record))
 						if(params["id"])
 							if(!valid_record_change(usr, "delete", null, active_security_record))
 								return
-							GLOB.data_core.removeMajorCrime(active_general_record.fields["id"], params["id"])
+							GLOB.data_core.removeComment(active_general_record.fields["id"], params["id"])
+
+				if("comment_add")
+					if(istype(active_general_record, /datum/data/record))
+						var/t1 = stripped_multiline_input("Add Comment:", "Secure. records", null, null)
```
Comment (Bibby0110) [inline code/game/machinery/computer/security.dm]: give this a proper and descriptive name?
Remove the two null statements at the end unless it's on purpose that you're removing the max message length (1024)
Add usr for consistency or just wipe it from the rest of the file. (It's mixed if consoles use the usr, but everything else does for some reason)
```suggestion
						var/t1 = stripped_multiline_input(usr, "Add Comment:", "Security Records")
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -414,54 +414,35 @@
...
-													else if (!H.canUseHUD())
-														return
-													else if (!istype(H.glasses, /obj/item/clothing/glasses/hud/security) && !istype(H.getorganslot(ORGAN_SLOT_HUD), /obj/item/organ/cyberimp/eyes/hud/security))
-														return
-													var/crime = GLOB.data_core.createCrimeEntry(t1, t2, allowed_access, station_time_timestamp())
-													GLOB.data_core.addMajorCrime(R.fields["id"], crime)
-													investigate_log("New Major Crime: <strong>[t1]</strong>: [t2] | Added to [R.fields["name"]] by [key_name(usr)]", INVESTIGATE_RECORDS)
-													to_chat(usr, "<span class='notice'>Successfully added a major crime.</span>")
+									if(R)
+										var/t1 = stripped_input("Please input crime names:", "Security HUD", "", null)
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/human/human.dm]: variable names, t1 and t2 are just plain confusing

Comment (Bibby0110) [inline code/modules/mob/living/carbon/human/human.dm]: Remove trailing null if you want to enforce the max message length of 1024 characters

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -414,54 +414,35 @@
...
+										var/t2 = stripped_multiline_input("Please input crime details:", "Security HUD", "", null)
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/human/human.dm]: see above about trailing null parameter

File: code/datums/datacore.dm [hunk 4]
```diff
@@ -48,6 +56,14 @@
 	c.dataId = ++securityCrimeCounter
 	return c
 
+/datum/datacore/proc/createCommentEntry(commentText = "", author = "", time = "")
+	var/datum/data/comment/C = new /datum/data/comment
+	C.commentText = commentText
+	C.author = author
+	C.time = time != "" ? time : text("[] [], []", station_time_timestamp(), time2text(world.realtime, "MMM DD"), GLOB.year_integer+540)
```
Comment (Bibby0110) [inline code/datums/datacore.dm]:
```suggestion
	C.time = time ? time : text("[] [], []", station_time_timestamp(), time2text(world.realtime, "MMM DD"), GLOB.year_integer+540)
```

File: code/game/machinery/computer/security.dm [hunk 5]
```diff
@@ -627,39 +617,39 @@
...
+						if(!valid_record_change(usr, name, null, active_security_record))
+							return
+						var/crime = GLOB.data_core.createCommentEntry(t1, logged_in)
```
Comment (Bibby0110) [inline code/game/machinery/computer/security.dm]: wrong variable name

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -488,10 +467,8 @@
 												return
 											else if(!istype(H.glasses, /obj/item/clothing/glasses/hud/security) && !istype(H.getorganslot(ORGAN_SLOT_HUD), /obj/item/organ/cyberimp/eyes/hud/security))
 												return
-											var/counter = 1
-											while(R.fields[text("com_[]", counter)])
-												counter++
-											R.fields[text("com_[]", counter)] = text("Made by [] on [] [], []<BR>[]", allowed_access, station_time_timestamp(), time2text(world.realtime, "MMM DD"), GLOB.year_integer+540, t1)
+											var/comment = GLOB.data_core.createCommentEntry(t1, allowed_access)
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/human/human.dm]: change the t1 variable while you're at it

File: code/game/machinery/computer/security.dm [hunk 7]
```diff
@@ -217,29 +217,28 @@
...
-				major_crime["name"] = C.crimeName
-				major_crime["details"] = C.crimeDetails
-				major_crime["author"] = C.author
-				major_crime["time"] = C.time
-				major_crime["id"] = C.dataId
+			for(var/datum/data/comment/C in active_security_record.fields["comments"])
+				var/list/comment = list()
+				comment["content"] = C.commentText
+				comment["author"] = C.author
+				comment["time"] = C.time
```
Comment (Bibby0110) [inline code/game/machinery/computer/security.dm:238]: shows full time + date unlike crimes

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
Might want to look at reworking medical comments now that you've refactored the datacore to handle them properly.

**Review by SomeguyManperson — APPROVED** [review]

**Comment by Cartlord** [comment]
Based adam, bibby BTFO

**Pull request merged**